### PR TITLE
⭐️ add macos.systemExtensions as new resource

### PR DESCRIPTION
**Pull request by chris-rock**
List all macOS system extensions

```coffeescript
cnquery> macos.systemExtensions
macos.systemExtensions: [
  0: macos.systemExtension teamID="P3FGV63VK7" identifier="io.kandji.KandjiAgent.ESF-Extension" version="1.8.10" state="activated_enabled"
  1: macos.systemExtension teamID="X9E956P446" identifier="com.crowdstrike.falcon.Agent" version="7.17" state="activated_enabled"
  2: macos.systemExtension teamID="2MMRE5MTB8" identifier="com.obsproject.obs-studio.mac-camera-extension" version="30.1.2" state="activated_waiting_for_user"
  3: macos.systemExtension teamID="X9E956P446" identifier="com.crowdstrike.falcon.Agent" version="7.16" state="terminated_waiting_to_uninstall_on_reboot"
  4: macos.systemExtension teamID="P3FGV63VK7" identifier="io.kandji.KandjiAgent.ESF-Extension" version="1.8.9" state="terminated_waiting_to_uninstall_on_reboot"
  5: macos.systemExtension teamID="DE8Y96K9QP" identifier="com.cisco.anyconnect.macos.acsockext" version="4.9.06037" state="activated_enabled"
]
```

To list extensions that are not mdm managed:

```coffeescript
cnquery> macos.systemExtensions.where(mdmManaged == false )
macos.systemExtensions.where: [
  0: macos.systemExtension teamID="2MMRE5MTB8" identifier="com.obsproject.obs-studio.mac-camera-extension" version="30.1.2" state="activated_waiting_for_user"
  1: macos.systemExtension teamID="DE8Y96K9QP" identifier="com.cisco.anyconnect.macos.acsockext" version="4.9.06037" state="activated_enabled"
]
```

### CODE REVIEW

File: providers/os/resources/os.lr.manifest.yaml [hunk 1]
```diff
@@ -527,6 +529,22 @@ resources:
...
+      enabled: {}
+      identifier: {}
+      mdmManaged: {}
+      name: {}
+      state: {}
+      teamID: {}
+      uuid: {}
+      version: {}
+    is_private: true
+    min_mondoo_version: latest
```
Comment (tas50) [inline providers/os/resources/os.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

File: providers/os/resources/os.lr.manifest.yaml [hunk 2]
```diff
@@ -506,6 +506,8 @@ resources:
   macos:
     fields:
       globalAccountPolicies: {}
+      systemExtensions:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/os/resources/os.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/os/resources/os.lr [hunk 3]
```diff
@@ -1400,6 +1402,30 @@ windows {
...
+  identifier string
+  // System extension unique identifier
+  uuid string
+  // Version of the system extension
+  version string
+  // Categories of the system extension
+  categories []string
+  // State of the system extension
+  state string
+  // Indicates whether the system extension is enabled
```
Comment (misterpantz) [inline providers/os/resources/os.lr]:
```suggestion
  // Whether the system extension is enabled
```

File: providers/os/resources/os.lr [hunk 4]
```diff
@@ -1400,6 +1402,30 @@ windows {
...
+  enabled() bool
+  // Indicates whether the system extension is active
```
Comment (misterpantz) [inline providers/os/resources/os.lr]:
```suggestion
  // Whether the system extension is active
```

File: providers/os/resources/os.lr [hunk 5]
```diff
@@ -1400,6 +1402,30 @@ windows {
...
+  active() bool
+  // Team identifier of the system extension
+  teamID string
+  // Path to the system extension
+  bundlePath string
+  // Indicates whether the system extension is MDM managed
```
Comment (misterpantz) [inline providers/os/resources/os.lr]:
```suggestion
  // Whether the system extension is MDM managed
```

**Review by misterpantz — COMMENTED** [review]
Good help text, @chris-rock! I just updated the boolean descriptions to fit our standard... hope to get them all consisten eventually

**Review by imilchev — APPROVED** [review]

**Pull request merged**